### PR TITLE
rest + soap request examples

### PR DESCRIPTION
**Pull request by Taniann**

### CODE REVIEW

File: src/main/java/ua/mateacademy/tania/lesson12/json/MateGroup.java [hunk 1]
```diff
@@ -107,21 +107,21 @@ public static MateGroup mateGroupExampleCreator() {
...
         List<Person> students = new ArrayList<>();
-        students.add(new Student("Iza", "Rabinovish", 1997));
-        students.add(new Student("Abram", "Kolom", 1998));
-        students.add(new Teacher("Moisha", "Stefan", 1990, 666));
+        students.add(new Student(1,"Iza", "Rabinovish", 1997));
+        students.add(new Student(2,"Abram", "Kolom", 1998));
+        students.add(new Teacher(3,"Moisha", "Stefan", 1990, 666));
         mateGroup.setStudents(students);
 
         List<Laptop> laptops = new ArrayList<>();
```
Comment (MateAcademy) [inline src/main/java/ua/mateacademy/tania/lesson12/json/MateGroup.java:127]: Please explain what is this link? Where is it used?

File: src/main/java/ua/mateacademy/tania/lesson12/json/Person.java [hunk 2]
```diff
@@ -50,9 +53,40 @@ public void setYearOfBorn(int yearOfBorn) {
...
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    public void update(Person person) {
+        this.name = person.getName();
+        this.surname = person.getSurname();
+        this.yearOfBorn = person.getYearOfBorn();
+    }
```
Comment (MateAcademy) [inline src/main/java/ua/mateacademy/tania/lesson12/json/Person.java:91]: Why you override equals, hashCode, when use method update?

File: src/main/java/ua/mateacademy/tania/lesson13/rest/MateGroupService.java [hunk 3]
```diff
@@ -8,7 +8,9 @@
  * Created by Tania Nebesna on 29.01.2019.
  */
 public interface MateGroupService {
-    public Response getMateGroup();
+    Response getMateGroup();
 
-    public Response addStudent(int groupId, Person person);
+    Response addStudent(int groupId, Person person);
+
+    Response updateStudent(Person person);
```
Comment (MateAcademy) [inline src/main/java/ua/mateacademy/tania/lesson13/rest/MateGroupService.java]: Where the teacher is changing?

File: src/main/java/ua/mateacademy/tania/lesson13/soap/MateGroupService.java [hunk 4]
```diff
@@ -17,9 +17,13 @@
...
-    public MateGroup getMateGroup();
+    MateGroup getMateGroup();
 
     @WebMethod
-    public MateGroup addStudents(List<Person> persons);
+    MateGroup addStudents(List<Person> persons);
+
+    @WebMethod
+    MateGroup updateStudent(Person person);
+
```
Comment (MateAcademy) [inline src/main/java/ua/mateacademy/tania/lesson13/soap/MateGroupService.java:28]: Where the teacher is changing in soap?

File: src/main/java/ua/mateacademy/tania/lesson13/rest/MateGroupServiceImpl.java [hunk 5]
```diff
@@ -40,4 +38,23 @@ public Response addStudent(@PathParam("groupId") int groupId, Person person) {
         }
         return Response.status(NOT_FOUND).build();
     }
+
+    @Override
+    @PUT
+    @Path("/update")
```
Comment (SerhiiPasko) [inline src/main/java/ua/mateacademy/tania/lesson13/rest/MateGroupServiceImpl.java]: Bad path value. Better write "student". In this path (same) you can insert, update, delete and get students, but by different operations

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
My task was:ДЗ - додати по методу до реста і соапа PUT який буде змінювати вчителя (також реквести постмана). Main goal - you must create new methods with specific signature

**Review by SerhiiPasko — APPROVED** [review]
Good

**Review by MateAcademy — APPROVED** [review]
yes

**Pull request merged**